lib/generator.js

```diff
@@ -44,9 +44,10 @@ Generator.getDomainList = function() {
 
 Generator.compile = function (fn) {
   var templates = this.getTemplatesSync();
-  // Remove /'s from the string representation to allow language-specific
-  // regexp delimiters to be used.
-  var regexp = Generator.getEmailRegexp().toString().slice(1, -1);
+  // Remove /'s and ^$ from the string representation to allow
+  // language-specific regexp delimiters and start/end-of-string anchors to be
+  // used.
+  var unanchoredRegexp = Generator.getEmailRegexp().toString().slice(2, -2);
```

Comment:
I have to admit I don't recall why I was drawn to the `getRegexp().toString().slice(2, -2);` solution rather than one like you suggest. I agree with your suggestion though, so I'll fix the PR.